sqlmesh/core/snapshot/evaluator.py

```diff
@@ -829,15 +844,47 @@ def _migrate_snapshot(
         if not needs_migration:
             return
 
-        tmp_table_name = snapshot.table_name(is_deployable=False)
+        evaluation_strategy = _evaluation_strategy(snapshot, adapter)
+
         target_table_name = snapshot.table_name()
-        _evaluation_strategy(snapshot, adapter).migrate(
-            target_table_name=target_table_name,
-            source_table_name=tmp_table_name,
-            snapshot=snapshot,
-            snapshots=parent_snapshots_by_name(snapshot, snapshots),
-            allow_destructive_snapshots=allow_destructive_snapshots,
-        )
+        if adapter.table_exists(target_table_name):
```

Comment:
so this will only be true when it's been deployed to prod, because otherwise only the temp table will exist?

Reply:
Exactly right